libs/api/domains/rsk-company-info/src/lib/graphql/rsk-company-info.service.ts

```diff
@@ -0,0 +1,164 @@
+import { Inject, Injectable, NotFoundException } from '@nestjs/common'
+import { RskCompany } from './models/rskCompany.model'
+import { CompanyApi } from '@island.is/clients/rsk-company-info'
+import { RskCompanyFormOfOperation } from './models/rskCompanyFormOfOperation.model'
+import { RskCompanyVat } from './models/rskCompanyVat.model'
+import { RskCompanyAddress } from './models/rskCompanyAddress.model'
+import { RskCompanyRelatedParty } from './models/rskCompanyRelatedParty.model'
+import { RskCompanyClassification } from './models/rskCompanyClassification.model'
+import { RskCompanySearchItems } from './models/rskCompanySearchItems.model'
+import { fromCursorHash, toCursorHash } from '../rsk-company-info.utils'
+import { PageResult } from './types/pageResult'
+import { LOGGER_PROVIDER } from '@island.is/logging'
+import type { Logger } from '@island.is/logging'
+
+@Injectable()
+export class RskCompanyInfoService {
+  constructor(
+    private rskCompanyInfoApi: CompanyApi,
+    @Inject(LOGGER_PROVIDER)
+    private logger: Logger,
+  ) {}
+
+  async getCompanyInformationWithExtra(
+    nationalId: string,
+  ): Promise<RskCompany> {
+    this.logger.debug(`Service getting company by nationalId ${nationalId}`)
+    const company = await this.rskCompanyInfoApi.v1FyrirtaekjaskraSsidGet({
+      ssid: nationalId,
+    })
+    this.logger.debug(`Company in service ${company.toString()}`)
+    return {
+      nationalId: company.kennitala,
+      name: company.nafn,
+      dateOfRegistration: new Date(company.skrad),
+      status: company.stada,
+      companyInfo: {
+        formOfOperation: company.rekstrarform?.map((item) => {
+          return {
+            type: item.tegund,
+            name: item.heiti,
+            suffix: item.vidskeyti,
+          } as RskCompanyFormOfOperation
+        }),
+        address: company.heimilisfang?.map((item) => {
+          return {
+            streetAddress: item.heimilisfang1,
+            streetAddress2: item.heimilisfang2,
+            postalCode: item.postnumer,
+            city: item.sveitarfelag,
+            cityNumber: item.sveitarfelagsnumer,
+            country: item.land,
+          } as RskCompanyAddress
+        }),
+        relatedParty: company.tengdirAdilar?.map((item) => {
+          return {
+            type: item.tegund,
+            nationalId: item.kennitala,
+            name: item.nafn,
+          } as RskCompanyRelatedParty
+        }),
+        vat: company.virdisaukaskattur?.map((item) => {
+          return {
+            vatNumber: item.vskNumer,
+            dateOfRegistration: new Date(item.skrad),
+            status: item.stada,
+            dateOfDeregistration: new Date(item.afskraning),
```

Comment:
Are these Date fields always set? Otherwise we may end up with some weird dates.